src/utils/headersToSchema.ts

```diff
@@ -0,0 +1,20 @@
+import { HttpHeader, ObjectSchema, Property } from "@azure-tools/codemodel";
+
+export function headersToSchema(
+  headers: HttpHeader[] | undefined,
+  operationFullName: string
+): ObjectSchema | undefined {
+  if (!headers || !headers.length) {
+    return undefined;
+  }
+  const headersSchema = new ObjectSchema(`${operationFullName}Headers`, "");
+
+  headers.forEach(({ header, schema }) => {
+    if (!headersSchema.properties) {
+      headersSchema.properties = [];
+    }
+    headersSchema.properties.push(new Property(header, "", schema));
```

Comment:
Can headers have a description?

Reply:
I think so, I'll extract it from the schema and add it, also above